docs/advanced/signal_generators.md

```diff
@@ -0,0 +1,558 @@
+```
+
+The AWG class has some methods specific to it. `generate_waveform` allows for a waveform name from the waveform list
+to be provided, instead of a function. The method is also distinctly different from generate function as it relies on a sample
+rate also being provided to actually generate the waveform.
+
+AWGs have access to the following functions:
+`SIN`, `SQUARE`, `RAMP`, `TRIANGLE`, `DC`, `CLOCK`
+
+### AWG5K/AWG7K
+
+The AWG5K/7K series are signal sources focused on waveform generation which operate on Windows XP (Windows 7 for model C).
+They accept communication through LAN and GPIB interfaces.
+
+`set_output_signal_path` is uniquely defined within the AWG5K and AWG7K classes, as it will set the value for
+`AWGCONTROL:DOUTPUTx:STATE`, which is a unique option not seen in the other AWGs.
+
+`set_offset` is conditioned to make sure that the AWG output signal path is not DIR, as the VISA query will time
+out otherwise.
+
+```{note}
+Operation complete commands will always return 1 on the AWG5K/7K series.
+```
+
+```{caution}
+All waveforms must be of the same length on requesting AWGCONTROL:RUN.
+```
+
+#### AWG5K, AWG5KB, AWG5KC
+
+#### Constraints:
+
+The AWG5K series offers a upper sample rate range from 600 MS/s to 1.2 GS/s dependent on the model number. The amplitude range is dependent on
+the signal output path, and if the direct option is selected, it will reduce the amplitude range to 20mV to 0.6 V and the
+offset to 0. Otherwise, the amplitude ranges from 20 mV to 2.25 V and an offset range of plus or minus 2.25 V.
+
+```{table} AWG5K Constraints
+---
+widths: auto
+width: 50%
+align: center
+---
+|                   | 500x/B/C   | 501x/B/C    | 5xxx/B/C DIR |
+| ----------------- | ---------- | ----------- | ------------ |
+| Sample Rate (S/s) | 10M–600M   | 10M–1.2M    | "            |
+| Amplitude (V)     | 20m–2.25   | 20m–2.25    | 20m-0.6      |
+| Offset (V)        | -2.25–2.25 | -2.25–2.25  | N/A          |
+
+```
+
+```{note}
+AWG5K's have digitized outputs on the rear of the device.
+```
+
+#### AWG7K, AWG7KB, AWG7KC
+
+#### Constraints:
+
+The AWG7K series functions identically to the [AWG5K](#awg5k-awg5kb-awg5kc) series, excluding the higher sample rate and lower amplitude and offset range.
+
+```{table} AWG7K Constraints
+---
+widths: auto
+width: 50%
+align: center
+---
+|                   | 705x     | 710x     | 706xB    | 712xB/C  | 708xC    |
+| ----------------- | -------- | -------- | -------- | -------- | -------- |
+| Sample Rate (S/s) | 10M–5G   | 10M–10G  | 10M–6G   | 10M–12G  | 10M–8G   |
+| Amplitude (V)     | 50m–2    | 50m–2    | 50m–2    | 50m–2    | 50m–2    |
+| Offset (V)        | -0.5–0.5 | -0.5–0.5 | -0.5–0.5 | -0.5–0.5 | -0.5–0.5 |
+
+```
+
+The AWG7K also includes varying options which directly effect these ranges, such as option 02 and 06. These options will enforce the
+output signal path to always be direct.
```

Comment:
Maybe add the Enum here and just update the glossary.

```suggestion
output signal path to always be `DIR`.
```

Reply:
Although DIR is the convention used here and functions similarily between the AWG5k7k and AWG70k, the commands used maay be different. I have changes the paragraph to explain this nuance instead.